source/slang/emit.cpp

```diff
@@ -5612,9 +5612,20 @@ struct EmitVisitor
                             {
                             default: Emit("rgba");  break;
 
-                            // TODO: GLSL doesn't actually seem to support 3-component formats
-                            // universally, so this might cause problems.
-                            case 3:  Emit("rgb");   break;
+                            case 3:
+                            {
+                                // TODO: GLSL doesn't actually seem to support 3-component formats
+                                // universally so for now we are going to default to rgba
+                                //
+                                // It seems that SPIR-V can support such formats, so with direct SPIR-V
+                                // path, and/or with support glslang could support in the future.
+                                //
+                                // Should this emit a warning? That it requires something different bound
+                                // than specified in source, and that could be pretty confusing.
```

Comment:
We need a consistent plan for what to do about this stuff. The simplest answer will be to drop these qualifiers for Vulkan targets, if they really aren't needed (at which point there's no need for warnings).

Then for OpenGL targets it would seem reasonable to have well-defined rules for inferring the format (and just document that 3-component formats map to 4-component formats, but that shouldn't matter because the API wouldn't let the user allocate those 3-component formats anyway...), and add an attribute for specifying the format manually if you really want to override our inference (e.g., to specify `r11fg11fb10f`).